src/kakarot/instructions/system_operations.cairo

```diff
@@ -924,6 +1051,12 @@ namespace SelfDestructHelper {
             sub_context=ctx.sub_context,
             destroy_contracts_len=0,
             destroy_contracts=empty_destroy_contracts,
+            events_len=0,
+            events=cast(0, model.Event*),
```

Comment:
Probably should just get rid of all the casts because it can cause unexpected memory bugs.